infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -404,23 +452,26 @@ private void handleFlushCacheRequest(CorfuMsg msg, ChannelHandlerContext ctx, IS
      */
     @Override
     public void sealServerWithEpoch(long epoch) {
-        CorfuPayloadMsg<Long> msg = new CorfuPayloadMsg<>();
-        msg.setEpoch(epoch);
-        msg.setPriorityLevel(PriorityLevel.HIGH);
+        RequestMsg batchProcessorMsg = getRequestMsg(
+                getHighPriorityHeaderMsg(HeaderMsg.newBuilder().setEpoch(epoch).build()),
+                getSealRequestMsg(epoch)
+        );
+
         try {
-            batchWriter.addTask(SEAL, msg).join();
+            batchWriter.addTask(BatchWriterOperation.Type.SEAL, batchProcessorMsg).join();
         } catch (CompletionException ce) {
             if (ce.getCause() instanceof WrongEpochException) {
                 // The BaseServer expects to observe this exception,
                 // when it happens, so it needs to be unwrapped
                 throw (WrongEpochException) ce.getCause();
             }
         }
+
         log.info("LogUnit sealServerWithEpoch: sealed and flushed with epoch {}", epoch);
     }
 
     @Override
-    public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
+    public boolean isServerReadyToHandleMsg(RequestMsg request) {
```

Comment:
Note: this method can be removed since this is the default behaviour of `AbstractServer`.